inbm/dispatcher-agent/dispatcher/sota/tiber_util.py

```diff
@@ -0,0 +1,161 @@
+"""
+    ORAS tool will be called by dispatcher to perform the image downloading in TiberOS.
+
+    Copyright (C) 2017-2024 Intel Corporation
+    SPDX-License-Identifier: Apache-2.0
+"""
+import logging
+import os
+import requests
+from requests import HTTPError
+from requests.exceptions import ProxyError, ChunkedEncodingError, ContentDecodingError, ConnectionError
+
+import shlex
+from urllib.parse import urlsplit
+from typing import Optional, Any
+from inbm_common_lib.utility import CanonicalUri
+from dispatcher.packagemanager.package_manager import verify_source
+from ..packagemanager.irepo import IRepo
+from ..dispatcher_broker import DispatcherBroker
+from .constants import RELEASE_SERVER_TOKEN_PATH
+from .sota_error import SotaError
+
+logger = logging.getLogger(__name__)
+
+
+def tiber_download(dispatcher_broker: DispatcherBroker, uri: CanonicalUri,
+                   repo: IRepo, username: Optional[str], password: str, umask: int) -> None:
+    """Downloads files and places capsule file in path mentioned by manifest file.
+
+    @param dispatcher_broker: DispatcherBroker object used to communicate with other INBM services
+    @param uri: URI of the source location
+    @param repo: repository for holding the download
+    @param username: username to use for download
+    @param password: password to use for download
+    @param umask: file permission mask
+    @raises SotaError: any exception
+    """
+    dispatcher_broker.telemetry(f'Package to be fetched from {uri.value}')
+    dispatcher_broker.telemetry(
+        'Checking authenticity of package by checking signature and source')
+
+    if not isinstance(uri, CanonicalUri):
+        raise SotaError("Internal error: uri improperly passed to download function")
+
+    source = uri.value[:-(len(uri.value.split('/')[-1]) + 1)]
+    file_name = os.path.basename(urlsplit(uri.value).path)
+    logger.debug(f"source: {source}, filename: {file_name}")
+
+    verify_source(source=source, dispatcher_broker=dispatcher_broker)
+    dispatcher_broker.telemetry('Source Verification check passed')
+
+    if password:
```

Comment:
there's some confusion in terminology here between password and token--let's stick to one of them

Reply:
I think token is more suitable here as we are getting the JWT token to download the file. Will change the password to token in this module.